codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpProtocolClientGenerator.kt

```diff
@@ -245,18 +245,18 @@ abstract class HttpProtocolClientGenerator(
         val hasOutputStream = outputShape.map { it.hasStreamingMember(ctx.model) }.orElse(false)
         val inputVariableName = if (inputShape.isPresent) "input" else KotlinTypes.Unit.fullName
 
-        writer
-            .write(
-                """val rootSpan = config.tracer.createRootSpan("#L-${'$'}{op.context.#T}")""",
-                op.id.name,
-                RuntimeTypes.HttpClient.Operation.sdkRequestId,
-            )
-            .withBlock(
-                "return #T.#T(rootSpan) {",
+        writer.withBlock(
+                """return config.tracer.#T(#S) { span ->""",
                 "}",
-                RuntimeTypes.KotlinCoroutines.coroutineContext,
-                RuntimeTypes.Tracing.Core.withRootTraceSpan,
+                RuntimeTypes.Tracing.Core.withSpan,
+                op.id.name
             ) {
+                write("span.#T(#S, op.context.#T!!)", RuntimeTypes.Tracing.Core.setAttribute, "rpc.method", RuntimeTypes.SmithyClient.operationName)
+                write("span.#T(#S, ServiceId)", RuntimeTypes.Tracing.Core.setAttribute, "rpc.service")
+                // FIXME - set client name
```

Comment:
**Question**: Is it still your intent to allow setting a client name to disambiguate amongst multiple clients for the same service? How do you think that will be sent downstream to probes?